### PR TITLE
Add `XeTLA` FA backward implementation to benchmark

### PR DESCRIPTION
**Pull request by ESI-SYD**
Implementation comes from [IPEX](https://github.com/intel-innersource/frameworks.ai.pytorch.ipex-gpu/blob/e0026ba2e9e5d3d1601059334b4d013ec77a4625/csrc/gpu/aten/operators/xetla/kernels/SDP/fmha_backward.cpp) 

https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/11089057652

### CODE REVIEW

**Review by chengjunlu — APPROVED** [review]

File: benchmarks/xetla_kernel/python_main.cpp [hunk 1]
```diff
@@ -147,6 +148,75 @@ void flash_attn(const at::Tensor &q, const at::Tensor &k, const at::Tensor &v,
...
+  CHECK_INPUT(grad_q_tmp);
+  CHECK_INPUT(grad_query);
+  CHECK_INPUT(grad_key);
+  CHECK_INPUT(grad_value);
+  CHECK_INPUT(grad_bias);
+
+#ifdef USE_IPEX
+  RECORD_FUNCTION("xetla fa", {num_batches, num_heads, head_size, num_queries,
+                               num_keys, bias_strideB, bias_strideN,
+                               bias_strideF, attn_mask_padding});
```
Comment (anmyachev) [inline benchmarks/xetla_kernel/python_main.cpp:195]:
```suggestion
  RECORD_FUNCTION("xetla fa", {});
```

**Review by whitneywhtsang — COMMENTED** [review]
Can we run it on CI?

**Review by chengjunlu — APPROVED** [review]

**Comment by ESI-SYD** [comment]
> Can we run it on CI and check if the result can be compared with torch?

CI tutorial seems not ready for backward (include torch's implication) yet.

File: benchmarks/xetla_kernel/CMakeLists.txt [hunk 2]
```diff
@@ -2,7 +2,10 @@
 find_package(XeTLALibrary REQUIRED)
 set(CMAKE_CXX_STANDARD 20)
 
-set(XETLA_KERNEL_FLAGS ${XETLA_KERNEL_FLAGS} -fsycl)
+set(XETLA_KERNEL_FLAGS ${XETLA_KERNEL_FLAGS}
+  -fsycl
+  -fsycl-device-code-split=per_kernel
```
Comment (ZzEeKkAa) [inline benchmarks/xetla_kernel/CMakeLists.txt:7]: What does it change?

Comment (ESI-SYD) [reply benchmarks/xetla_kernel/CMakeLists.txt:7]: This porting adopts option from https://github.com/intel-innersource/frameworks.ai.pytorch.ipex-gpu/blob/master/csrc/gpu/aten/operators/xetla/kernels/compile_library.cmake#L6

This flag means performing sycl kernel splitting, helps solving RuntimeError below introduced by this porting. 

No perf regression for xetla in my local env.

```bash
RuntimeError: The program was built for 1 devices
Build program log for 'Intel(R) Data Center GPU Max 1100':
 -11 (PI_ERROR_BUILD_PROGRAM_FAILURE)
```

**Pull request merged**